File: TheMovie/app/src/main/java/com/nyinyi/nw/themovie/activity/SearchresultActivity.java
package com.nyinyi.nw.themovie.activity;

import android.os.Bundle;
import android.widget.Toast;

import com.nyinyi.nw.themovie.R;
import com.nyinyi.nw.themovie.adapter.RecyclerAdapter;
import com.nyinyi.nw.themovie.event.DataEvent;
import com.nyinyi.nw.themovie.network.RetrofitDataAgentImpl;

import org.greenrobot.eventbus.EventBus;
import org.greenrobot.eventbus.Subscribe;
import org.greenrobot.eventbus.ThreadMode;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.GridLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import butterknife.BindView;
import butterknife.ButterKnife;

public class SearchresultActivity extends AppCompatActivity {

    @BindView(R.id.rv_movie)
    RecyclerView recycler;

    private String search_name;
    private RecyclerAdapter radapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_searchresult);

        ButterKnife.bind(this, this);
        search_name = getIntent().getStringExtra("search_name");

        if (getSupportActionBar() != null) getSupportActionBar().setTitle(search_name);


        radapter = new RecyclerAdapter(getApplicationContext());
        setupRecycler();
        RetrofitDataAgentImpl.getInstance().loadsearchMovie(search_name);


    }

    private void setupRecycler() {
        recycler.setHasFixedSize(true);
        recycler.setLayoutManager(new GridLayoutManager(getApplicationContext(), 3));
        recycler.setAdapter(radapter);
    }

    @Override
    public void onStart() {
        super.onStart();

        EventBus.getDefault().register(this);

    }

    @Override
    public void onStop() {
        super.onStop();

        EventBus.getDefault().unregister(this);

    }


    @Subscribe(threadMode = ThreadMode.MAIN)
    public void searchMovieLoadedEvent(DataEvent.SearchMovieLoadEvent event) {
        radapter.setsearchdatalist(event.getSearchVOList());
    }

    @Subscribe(threadMode = ThreadMode.MAIN)
    public void errorEvent(DataEvent.ErrorLoadedEvent event) {
        Toast.makeText(getApplicationContext(), event.getMessage(), Toast.LENGTH_SHORT).show();
    }

}

File: TheMovie/app/src/main/java/com/nyinyi/nw/themovie/activity/SearchresultActivity.kt
package com.nyinyi.nw.themovie.activity

import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.GridLayoutManager
import androidx.recyclerview.widget.RecyclerView
import butterknife.BindView
import butterknife.ButterKnife
import com.nyinyi.nw.themovie.R
import com.nyinyi.nw.themovie.adapter.RecyclerAdapter
import com.nyinyi.nw.themovie.event.DataEvent
import com.nyinyi.nw.themovie.network.RetrofitDataAgentImpl
import org.greenrobot.eventbus.EventBus
import org.greenrobot.eventbus.Subscribe
import org.greenrobot.eventbus.ThreadMode

class SearchresultActivity : AppCompatActivity() {

    @BindView(R.id.rv_movie)
    internal var recycler: RecyclerView? = null

    private var search_name: String? = null
    private var radapter: RecyclerAdapter? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_searchresult)

        ButterKnife.bind(this, this)
        search_name = intent.getStringExtra("search_name")

        if (supportActionBar != null) supportActionBar!!.setTitle(search_name)


        radapter = RecyclerAdapter(applicationContext)
        setupRecycler()
        RetrofitDataAgentImpl.instance.loadsearchMovie(search_name.toString())


    }

    private fun setupRecycler() {
        recycler!!.setHasFixedSize(true)
        recycler!!.layoutManager = GridLayoutManager(applicationContext, 3)
        recycler!!.adapter = radapter
    }

    public override fun onStart() {
        super.onStart()

        EventBus.getDefault().register(this)

    }

    public override fun onStop() {
        super.onStop()

        EventBus.getDefault().unregister(this)

    }


    @Subscribe(threadMode = ThreadMode.MAIN)
    fun searchMovieLoadedEvent(event: DataEvent.SearchMovieLoadEvent) {
        radapter!!.setsearchdatalist(event.searchVOList)
    }

    @Subscribe(threadMode = ThreadMode.MAIN)
    fun errorEvent(event: DataEvent.ErrorLoadedEvent) {
        Toast.makeText(applicationContext, event.message, Toast.LENGTH_SHORT).show()
    }

}
